api/src/services/authentication.ts

```diff
@@ -160,10 +160,25 @@ export class AuthenticationService {
 			}
 		}
 
-		const payload = {
+		let payload = {
 			id: user.id,
 		};
 
+		const [customClaims] = await emitter.emitAsync('auth.jwt.before', payload, {
```

Comment:
This will only destructure the first custom claim created in a hook. If the hook is multiple times, we should merge the result of every hook execution, and use that as the custom payload.

Reply:
Yes, that is right! I wasn't aware that a hook could be executed multiple times 😅 I will have a reducer